### PR TITLE
add anti-affinity to pods so that shards don't get scheduled on the s…

### PR DESCRIPTION
**Pull request by QuantumEnigmaa**
Towards https://github.com/giantswarm/giantswarm/issues/27744

This PR:

- adds anti-affinity in the Prometheus CR so that prometheus-agent shards don't get scheduled on the same node if possible.

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Test on Workload cluster.

### CODE REVIEW

**Comment by QuantumEnigmaa** [comment]
Tried to test it on a WC but since one needs 1M series to spawn a shard in addition to the prometheus-agent it requires a huge amount of pods. 
Even with almost 3000 pods from a single deployment, 15 worker nodes with 60 CPUs and 230Go of RAM I was only nearing 500K series so I decided to stop testing there.

File: helm/prometheus-agent/charts/prometheus-agent/templates/prometheus.yaml [hunk 1]
```diff
@@ -8,6 +8,18 @@ metadata:
   name: {{ include "name" . }}
   namespace: {{ .Release.Namespace }}
 spec:
+  affinity:
+    podAntiAffinity:
+      preferredDuringSchedulingIgnoredDuringExecution:
+      - weight: 100
+        podAffinityTerm:
+          labelSelector:
+            matchExpressions:
```
Comment (QuentinBisson) [inline helm/prometheus-agent/charts/prometheus-agent/templates/prometheus.yaml:17]: Should that not be based on the hostname instead?

**Comment by QuentinBisson** [comment]
If you want to test, you can apply the change manually on deu01 :)

**Comment by QuantumEnigmaa** [comment]
> If you want to test, you can apply the change manually on deu01 :)

Am I allowed to ? :O 
This will affect the customer

File: helm/prometheus-agent/charts/prometheus-agent/templates/prometheus.yaml (hunk 1 above)
Comment (QuantumEnigmaa) [reply helm/prometheus-agent/charts/prometheus-agent/templates/prometheus.yaml:17]: yes, my bad

**Comment by QuentinBisson** [comment]
This would only recreate the pods right? We would still have metrics. Let's not make it a regular thing but we do not have that much of a choice :(

**Comment by QuantumEnigmaa** [comment]
In any case, I was able to see that the agent did have its affinity set and since shards are basically using the same Prometheus CR it shouldn't change much

**Comment by QuentinBisson** [comment]
Well those are different statefulset for each shard so I'd validate that this actually works

**Review by QuentinBisson — APPROVED** [review]
If they are indeed on different nodes, LGTM

**Comment by QuantumEnigmaa** [comment]
Sorry forgot to mention that I tested it on `deu01` and the shards are indeed created on different nodes.

**Pull request merged**